src/scripts/main.js

```diff
@@ -1,3 +1,332 @@
 'use strict';
 
-// write code here
+const tableHeaders = document.querySelectorAll('th');
+const rowsArray = Array.from(document.querySelectorAll('tbody tr'));
+const sortDirection = {};
+
+// tableHeaders.forEach((header, index) => {
+//   sortDirection[index] = true;
+
+//   header.addEventListener('click', (e) => {
+//     const sortedRows = rowsArray.sort((rowA, rowB) => {
+//       const cellA = rowA.cells[index].textContent.trim();
+//       const cellB = rowB.cells[index].textContent.trim();
+
+//       if (index === 4) {
+//         const salaryA = parseFloat(cellA.replace(/[$,]/g, ''));
+//         const salaryB = parseFloat(cellB.replace(/[$,]/g, ''));
+
+//         return sortDirection[index] ? salaryA - salaryB : salaryB - salaryA;
+//       } else {
+//         return sortDirection[index]
+//           ? cellA.localeCompare(cellB)
+//           : cellB.localeCompare(cellA);
+//       }
+//     });
+
+//     const tbody = document.querySelector('tbody');
+
+//     tbody.innerHTML = '';
+//     sortedRows.forEach((row) => tbody.appendChild(row));
+
+//     sortDirection[index] = !sortDirection[index];
+//   });
+// });
+
+function sortRows(index, isAscending, rowsArr) {
+  return rowsArr.sort((rowA, rowB) => {
+    const cellA = rowA.cells[index].textContent.trim();
+    const cellB = rowB.cells[index].textContent.trim();
+
+    if (index === 4) {
+      const salaryA = parseFloat(cellA.replace(/[$,]/g, ''));
+      const salaryB = parseFloat(cellB.replace(/[$,]/g, ''));
+
+      return isAscending ? salaryA - salaryB : salaryB - salaryA;
+    } else {
+      return isAscending
+        ? cellA.localeCompare(cellB)
+        : cellB.localeCompare(cellA);
+    }
+  });
+}
+
+function updateTable(sortedRows) {
```

Comment:
The function `updateTable` is re-querying the DOM for the `tbody` element each time it's called. Consider passing the `tbody` element as a parameter to this function to avoid unnecessary DOM queries.